src/main/kotlin/org/rust/ide/presentation/Utils.kt

```diff
@@ -66,8 +66,27 @@ fun getPresentationForStructure(psi: RsElement): ItemPresentation {
     return PresentationData(presentation, null, icon, textAttributes)
 }
 
+// Short presentation for display in the struct viewer
 private fun presentableName(psi: RsElement): String? {
     return when (psi) {
+        is RsAbstractable -> {
+            val o = psi.owner
+            if (o is RsAbstractableOwner.Impl) {
+                val typeName = o.impl.typeReference?.text ?: return null
+                val traitName = o.impl.traitRef?.text
+                buildString {
+                    append("<")
+                    if (traitName != null) {
+                        append("$traitName for ")
+                    }
+                    append(typeName)
+                    append(">::${psi.name}")
+                    append(typeParameterBounds(o.impl))
+                }
+            } else {
+                psi.name
+            }
+        }
```

Comment:
This change causes test failures. And I don't think it's reasonable to change structure view rendering. It's possible to change renderer of implementation variants with `NavigationGutterIconBuilder.setCellRenderer` (in `RsImplsLineMarkerProvider.collectSlowLineMarkers`, see `info` variable)

Reply:
Thanks, I changed it so that it displays the presentation of the abstractable's owner.